synapse/rest/client/room.py

```diff
@@ -1177,7 +1177,9 @@ async def on_GET(
         self, request: SynapseRequest, room_id: str
     ) -> Tuple[int, JsonDict]:
         requester = await self._auth.get_user_by_req(request)
-        await self._auth.check_user_in_room(room_id, requester.user.to_string())
```

Comment:
Should we make `check_user_in_room` private if it is not used anywhere else besides in `check_user_in_room_or_world_readable`?

Reply:
There are legit calls in some places, mainly in typing notifications servlets, because you can send typing notifications only in rooms you're in. This one was the only one that did not really made sense to me

Reply:
Got it! Wasn't 100% sure from the description.